api/v1/verticadb_webhook.go

```diff
@@ -1617,6 +1621,39 @@ func (v *VerticaDB) checkImmutableSubclusterInSandbox(oldObj *VerticaDB, allErrs
 	return allErrs
 }
 
+// checkImmutableStsName ensures the statefulset name of the subcluster stays constant
+func (v *VerticaDB) checkImmutableStsName(oldObj *VerticaDB, allErrs field.ErrorList) field.ErrorList {
+	// We have an annotation to control the sts name. We are not going to allow
+	// this annotation to be added to existing subclusters. Otherwise, it will
+	// regenerate a new statefulset and keep the old one around. The statefulset
+	// name can only be set for new subclusters.
+	oldScMap := oldObj.GenSubclusterMap()
+	newScMap := v.GenSubclusterMap()
+	scIndexMap := v.GenSubclusterIndexMap()
+	for scName, oldSc := range oldScMap {
+		// Find the subcluster in the new map.
+		newSc, found := newScMap[scName]
+		// Only concerned with changes of existing subclusters. So, we can skip if
+		// its not found.
+		if !found {
+			continue
+		}
+		oldStsName := oldSc.GetStatefulSetName(oldObj)
+		newStsName := newSc.GetStatefulSetName(v)
+		if oldStsName != newStsName {
+			scInx := scIndexMap[scName]
+			path := field.NewPath("spec").Child("subclusters").Index(scInx).
+				Child("annotations").Key(vmeta.StsNameOverrideAnnotation)
+			err := field.Invalid(path,
+				newStsName,
+				fmt.Sprintf("Unable to rename the statefulset of subcluster %q to %q after creation of the subcluster.",
```

Comment:
Does `renaming the statefulset of subcluster %q after creation of the subcluster is not allowed` sound better? I think we should make it clear that we prohibit sts name change.